R/StackedLearner.R

```diff
@@ -1,739 +1,166 @@
+#'   If set to \code{TRUE}, base models are saved on disc at the working directory. 
+#'   This setting saves memory when huge models are fitted but also might take longer. 
+#'   Later during prediction this models are loaded. Models are saved with the 
+#'   name "saved.models<stack.id>.<base.learner.id>.RData".
+#'   Note that it only works for train-predict procedures as well as for resampling using holdout. 
+#'   Applying outer cross validation will result in wrong predictions due to the 
+#'   fact that model names does not seperate between different resample iterations.
+#'   Default is \code{FALSE}
+#' @param save.preds [\code{logical(1)}]\cr 
+#'   If set to \code{FALSE} models will not contain predictions. This reduce the 
+#'   object size. Note that function \code{resampleStackedLearnerAgain} does not 
+#'   work if saving prediction is disabled. Default is \code{TRUE}.
+#' @references Wolpert, David H. "Stacked generalization." Neural networks 5.2 (1992): 241-259.
+#'   \url{http://www.machine-learning.martinsewell.com/ensembles/stacking/Wolpert1992.pdf}
+#' @references Caruana, Rich, et al. "Ensemble selection from libraries of models." 
+#'   Proceedings of the twenty-first international conference on Machine learning. 
+#'   ACM, 2004. \url{http://www.cs.cornell.edu/~caruana/ctp/ct.papers/caruana.icml04.icdm06long.pdf}
 #' @examples
+#' \dontrun{
 #'   # Classification
 #'   data(iris)
 #'   tsk = makeClassifTask(data = iris, target = "Species")
 #'   base = c("classif.rpart", "classif.lda", "classif.svm")
 #'   lrns = lapply(base, makeLearner)
 #'   lrns = lapply(lrns, setPredictType, "prob")
-#'   m = makeStackedLearner(base.learners = lrns,
-#'     predict.type = "prob", method = "hill.climb")
-#'   tmp = train(m, tsk)
-#'   res = predict(tmp, tsk)
+#'   stk = makeStackedLearner(method = "ensembleselection", base.learners = lrns, predict.type = "prob", parset = list(init = 1, metric = mmce))
+#'   res = resample(stk, tsk, cv5, mmce)
 #'
 #'   # Regression
 #'   data(BostonHousing, package = "mlbench")
 #'   tsk = makeRegrTask(data = BostonHousing, target = "medv")
 #'   base = c("regr.rpart", "regr.svm")
 #'   lrns = lapply(base, makeLearner)
-#'   m = makeStackedLearner(base.learners = lrns,
-#'     predict.type = "response", method = "compress")
-#'   tmp = train(m, tsk)
-#'   res = predict(tmp, tsk)
+#'   stk = makeStackedLearner(base.learners = lrns, predict.type = "response", method = "ensembleselection", parset = list(init = 1, metric = mae))
+#'   m = train(stk, tsk)
+#'   res = predict(m, tsk)
+#' }
 #' @export
-makeStackedLearner = function(base.learners, super.learner = NULL, predict.type = NULL,
-  method = "stack.nocv", use.feat = FALSE, resampling = NULL, parset = list()) {
 
+makeStackedLearner = function(id = "stack", method = "superlearner", base.learners, 
+  predict.type = NULL, resampling = NULL, super.learner = NULL, use.feat = FALSE, 
+  parset = list(), save.on.disc = FALSE, save.preds = TRUE) {
```

Comment:
the name `parset` is a little bit misleading, isn't it? maybe something like `par.vals` or `es.par.vals` (for ensamble selection).